src/main/java/org/bytedeco/javacpp/tools/Templates.java

```diff
@@ -47,8 +47,12 @@ static boolean notExists(String s) {
         return strip(s).length() == s.length();
     }
 
-    /** Split s at ::, but taking care of qualified template arguments */
-    static List<String> splitNamespace(String s) {
+    static class SplitResult extends ArrayList<String> {
+        String parameterList;
```

Comment:
This is ugly, please try to make the splitNamespace() method non-static and put that parameterList variable as a field of Templates instead.

Reply:
Something like this:
```java
class Templates {
    static ...  

    ArrayList<String> namespace;
    String parameters;

    Templates(String name) {
      // instead of splitNamespace()
   }
}
```

Reply:
`split` is a method which produces a list of string and an optional string, so I don't see what's wrong with creating a specific class for representing this result. I'd probably use a record if we were in Java 17. 
But I can do your way if you prefer. The only thing is that the `Templates` is ill-named for that usage, and there is no reason that it contains methods `strip`and `notExists`. And what if we want to add another template-related method that returns another kind of composite result ? 
What about creating a separate `Split` class similar to what you suggest and keep the abstract utility class `Templates` ?

Reply:
> What about creating a separate `Split` class similar to what you suggest and keep the abstract utility class `Templates` ?

Sure, let's name it TemplateList though?

Reply:
I remind you that this method just split something like `ns::name(int p)` to `["ns", "name"]` and `"(int p)"`. It's not a list of templates. It just support the presence of template arguments. 
Also do you prefer this class to be inner to `Templates` or top-level ?

Reply:
Ok, I'd just add everything to the existing ArrayList, including the content of parameterList (whether it's null or not). Why do you want to have it as a value out of the list?

Reply:
For a cleaner code and avoid juggling with indices. There are cases where we split something and don't care about parameters, or know there is no parameters, so we don't have to check  `parameterList`. We can even receive the result of `split` as a `List<String>`.

Reply:
Right, so we could have `splitNamespace(String name, boolean parameters = false)`, so by default it won't append anything, but if it's true it will either append null or a `(...)` string. What's wrong with that?

Reply:
Matter of coding style I guess.
I just pushed a commit with your suggestion. Do you prefer it this way ?

Reply:
Something like that yeah. We could also add a splitNamespaceParameters() to make this more explicit, that's coding style. Not creating an unnecessary new class/type isn't coding style, it's good practice. :)

Reply:
> Not creating an unnecessary new class/type isn't coding style, it's good practice. :)

That's highly debatable, when it affects readability/maintainability. But we won't debate here. We won't agree anyway !
Let's keep the commit as is.